error_handler.py

```diff
@@ -0,0 +1,114 @@
+import datetime
+import io
+import logging
+import traceback
+import urllib
+
+import discord
+from discord.ext import commands
+
+logger = logging.getLogger(__name__)
+
+
+def get_full_class_name(obj):
+    module = obj.__class__.__module__
+    if module is None or module == str.__class__.__module__:
+        return obj.__class__.__name__
+    return module + '.' + obj.__class__.__name__
+
+class NonBugError(Exception):
+    """When this is raised instead of a normal Exception, on_command_error() will not attach a traceback or github
+    link. """
+    pass
+
+# completely overkill error handler shamelessly ripped from MediaForge, a bot i wrote
+class ErrorHandlerCog(commands.Cog):
+    def __init__(self, bot):
+        self.bot: commands.Bot = bot
+        self.antispambucket = {}
+
+    @commands.Cog.listener()
+    async def on_command_error(self, ctx: commands.Context, commanderror: commands.CommandError):
+        async def dmauthor(*args, **kwargs):
+            try:
+                return await ctx.reply(*args, **kwargs)
```

Comment:
Does this actually use a direct message consistently? If not, does the code need to change, or does the method name need to change?

Reply:
code needs to change, not entirely sure what's going on there